src/gt4py/testing/suites.py

```diff
@@ -468,17 +460,25 @@ def _test_generation(cls, test, externals_dict):
 
         # Assert strict equality for Dawn backends
         if implementation.backend.startswith("dawn"):
-            assert all(
-                field_info.boundary == cls.global_boundaries[name]
-                for name, field_info in implementation.field_info.items()
-                if field_info is not None
-            )
+            for name, field_info in implementation.field_info.items():
+                if field_info is None:
+                    continue
+                for i, ax in enumerate("IJK"):
+                    assert (
+                        ax not in field_info.axes
+                        or ax == "K"
+                        or field_info.boundary[i] == cls.global_boundaries[name][i]
+                    )
         else:
-            assert all(
-                field_info.boundary >= cls.global_boundaries[name]
-                for name, field_info in implementation.field_info.items()
-                if field_info is not None
-            )
+            for name, field_info in implementation.field_info.items():
+                if field_info is None:
+                    continue
+                for i, ax in enumerate("IJK"):
+                    assert (
+                        ax not in field_info.axes
+                        or ax == "K"
+                        or field_info.boundary[i] >= cls.global_boundaries[name][i]
+                    )
```

Comment:
Am I seeing things or is there the exact same code for both if branches?

Reply:
The very last comparison is different, since dawn apparently can't handle the different halos I can make it cleaner